internal/explorerutil/store.go

```diff
@@ -0,0 +1,321 @@
+package explorerutil
+
+import (
+	"database/sql"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"path"
+	"strings"
+
+	"go.sia.tech/core/types"
+	"go.sia.tech/siad/v2/explorer"
+	_ "modernc.org/sqlite"
+)
+
+type SqliteStore struct {
+	db *sql.DB
+}
+
+func NewStore(dir string) (*SqliteStore, error) {
+	db, err := sql.Open("sqlite", path.Join(dir, "store.db"))
+	if err != nil {
+		return nil, err
+	}
+	if _, err := db.Exec(`
+CREATE TABLE elements (
+    id VARCHAR(128),
+    type VARCHAR(128),
+    data BLOB
+);
+
+CREATE TABLE chainstats (
+    id VARCHAR(128),
+    data BLOB
+);
+
+CREATE TABLE unspentElements (
+    id VARCHAR(128),
+    type VARCHAR(128),
+	address VARCHAR(128)
+);
+
+CREATE TABLE transactions (
+    id VARCHAR(128),
+    data BLOB
+);
+
+CREATE TABLE addressTransactions (
+    id VARCHAR(128),
```

Comment:
If the lib supports foreign keys, we should use them https://www.sqlite.org/foreignkeys.html